content/teknologi/altinnstudio/altinn-api/app-api/data-element/_index.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Data elements
+description: The App API to work with the data elements associated with an instance
+toc: true
+tags: [api]
+weight: 100
+---
+
+## Overview
+
+A Data element consist of two parts; its metadata document and the actual data blob. This API work primarily with the blob while keeping the metadata document updated.
+
+**basePath**
+```http
+{org}/{appname}/instances/{instanceOwnerPartyId}/{instanceGuid}/data
+```
+
+## Get data 
+
+Endpoint for downloading the data blob.
+
+```http
+GET basePath/{dataGuid}
+Accept: application/{xml/json}
+```
+
+The response will depend on the type of blob. There are currently 2 primary types; forms and attachments.
+
+### Response with form data
+
+The respons will either be a json or xml serialized version of the object model depending on the **Accept** header in the request. The value **application/xml** will result in an XML document and the value **application/json** will result inn a JSON document.
+
+### Response with instance attachment
+
+The respons will be a file stream. Content-Type will be the same as the original value given when the file was uploaded. The same is true for the file name.
+
+```http
+Content-Disposition: attachment; filename=cute_cat.png; filename*=UTF-8''cute_cat.png
+Content-Length: 16994
+Content-Type: image/png
+```
+
+## Upload data
+
+Endpoint for initially uploading a new data element on a specific instance.
+
+```http
+POST basePath?dataType={data type name}
+```
+
+The **dataType** parameter is required and should contain the name of one of the data types defined on the application. Data types with an **appLogic** property are linked to a form and will have data validation and calculation rules associated with them. Submitted data will be deserialized from **application/xml** or **application/json** based on the content-type of the request. Data types without an **appLogic** property will be handeled as any random file and streamed directly to storage.
+
+### Submitting form as application/json
```

Comment:
```suggestion
### Uploading form data as application/json
```

Maybe this would link it closer to the upload topic? Only a personal preference as submitting has other connotations for me.